### PR TITLE
MSSDK 1260 Migrate purchase component

### PR DESCRIPTION
**Pull request by izabelapeza**

### CODE REVIEW

File: src/api/Customer/getCustomerOffers.ts [hunk 1]
```diff
@@ -3,9 +3,9 @@ import fetchWithJWT from 'util/fetchHelper';
 import getApiURL from 'util/environmentHelper';
 import jwtDecode from 'jwt-decode';
 
-const getCustomerOffers = async () => {
+const getCustomerOffers = async (): Promise<{ items: unknown }> => {
```
Comment (iwonakulacz) [inline src/api/Customer/getCustomerOffers.ts:6]: items are always an array of objects

File: src/api/Customer/getCustomerOffers.ts [hunk 2]
```diff
@@ -3,9 +3,9 @@ import fetchWithJWT from 'util/fetchHelper';
...
   const API_URL = getApiURL();
-  const { customerId } = jwtDecode(getData('CLEENG_AUTH_TOKEN'));
+  const { customerId } = jwtDecode<{ customerId: string }>(getData('CLEENG_AUTH_TOKEN'));
```
Comment (iwonakulacz) [inline src/api/Customer/getCustomerOffers.ts:8]: customerId should be number here

File: src/api/Customer/getPaymentDetails.ts [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
-import fetchWithJWT from 'util/fetchHelper';
+import fetchWithJWT from "util/fetchHelper";
 import getApiURL from 'util/environmentHelper';
 import jwtDecode from 'jwt-decode';
 
-const getPaymentDetails = async () => {
+const getPaymentDetails = async (): Promise<{ paymentDetails: unknown }> => {
   const API_URL = getApiURL();
-  const { customerId } = jwtDecode(getData('CLEENG_AUTH_TOKEN'));
+  const { customerId } = jwtDecode<{ customerId: string }>(getData('CLEENG_AUTH_TOKEN'));
```
Comment (iwonakulacz) [inline src/api/Customer/getPaymentDetails.ts:8]: same here customerId: number

File: src/api/Customer/getPaymentDetails.ts [hunk 4]
```diff
@@ -1,11 +1,11 @@
 import { getData } from 'util/appConfigHelper';
-import fetchWithJWT from 'util/fetchHelper';
+import fetchWithJWT from "util/fetchHelper";
 import getApiURL from 'util/environmentHelper';
 import jwtDecode from 'jwt-decode';
 
-const getPaymentDetails = async () => {
+const getPaymentDetails = async (): Promise<{ paymentDetails: unknown }> => {
```
Comment (iwonakulacz) [inline src/api/Customer/getPaymentDetails.ts:6]: array of obejcts

File: src/components/Offer/Offer.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+import OfferCheckoutCard from '../OfferCheckoutCard';
+import { OfferProps } from './Offer.types';
+
+const Offer = ({ onSubmit, onPaymentComplete, t = k => k }: OfferProps) => {
+  const [coupon, setCoupon] = useState('');
+  const { trialAvailable } = useAppSelector(selectOffer).offer;
+  const {
+    totalPrice,
+    discount: { applied: discountApplied }
+  } = useAppSelector(selectOrder).order;
```
Comment (jkorczakcleeng) [inline src/components/Offer/Offer.tsx:30]: we could create two custom selectors one for 'order' and one for 'order.order' ;)

File: src/components/Offer/Offer.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        </main>
+      </StyledOfferWrapper>
+    );
+  }
+
+  return (
+    <StyledOfferWrapper>
+      <Header />
+      <main>
+        <>
```
Comment (jkorczakcleeng) [inline src/components/Offer/Offer.tsx:49]: we can remove React.Fragment

File: src/api/Customer/getCustomerOffers.ts [hunk 7]
```diff
@@ -3,9 +3,29 @@ import fetchWithJWT from 'util/fetchHelper';
 import getApiURL from 'util/environmentHelper';
 import jwtDecode from 'jwt-decode';
 
-const getCustomerOffers = async () => {
+type CustomersOffer = {
```
Comment (jkorczakcleeng) [inline src/api/Customer/getCustomerOffers.ts:6]: 🥇

File: src/components/ErrorPage/ErrorPage.types.ts [hunk 8]
```diff
@@ -0,0 +1,8 @@
+export type Errors =
```
Comment (jkorczakcleeng) [inline src/components/ErrorPage/ErrorPage.types.ts:1]: ❤️

File: src/components/Offer/Offer.types.ts [hunk 9]
```diff
@@ -0,0 +1,5 @@
+export type OfferProps = {
+  onSubmit: (str: string) => void;
+  onPaymentComplete: () => void;
+  t?: (str: string) => string;
```
Comment (jkorczakcleeng) [inline src/components/Offer/Offer.types.ts:4]:
```
  t: ReturnType<typeof useTranslation>['t'];

File: src/redux/orderSlice.ts [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+      if (isErrorMsg(err)) return rejectWithValue(err.message);
+      return err;
+    }
+  }
+);
+
+export const orderSlice = createSlice({
+  name: 'order',
+  initialState,
+  reducers: {},
```
Comment (jkorczakcleeng) [inline src/redux/orderSlice.ts:126]: we can remove empty reducers

File: src/redux/orderSlice.ts [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+  extraReducers: builder => {
+    builder.addCase(fetchCreateOrder.pending, state => {
+      state.loading = true;
+    });
+    builder.addCase(fetchCreateOrder.fulfilled, (state, { payload }) => {
```
Comment (jkorczakcleeng) [inline src/redux/orderSlice.ts:131]: we could try to add PayloadAction if it is possible in extraReducers

File: src/redux/orderSlice.ts (hunk 10 above)
Comment (izabelapeza) [reply src/redux/orderSlice.ts:126]: I had a TypeScript error, when I tried doing it :/ (That error: _Property 'reducers' is missing in type '{ name: "order"; initialState: InitialState; extraReducers: (builder: ActionReducerMapBuilder<InitialState>) => void; }' but required in type 'CreateSliceOptions<InitialState, SliceCaseReducers<InitialState>, "order">'._)

File: src/api/Customer/getCustomerOffers.ts (hunk 7 above)
Comment (izabelapeza) [reply src/api/Customer/getCustomerOffers.ts:6]: 🎉

File: src/redux/orderSlice.ts (hunk 11 above)
Comment (izabelapeza) [reply src/redux/orderSlice.ts]: I added PayloadAction to fulfilled actions, but I couldn't find information how to properly add it to rejected ones.

File: src/redux/publisherConfigSlice.ts [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+    enabled: boolean;
+  };
+  paymentMethodsConfiguration: {
+    card: {
+      name: string;
+      billingAddressRequired: boolean;
+      billingAddressMode: string;
+      brands: string[];
+      brandsConfiguration: {
+        visa: {
```
Comment (iwonakulacz) [inline src/redux/publisherConfigSlice.ts:19]: there can be other brands than 'visa'  also

File: src/redux/publisherConfigSlice.ts [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+      autoFocus: boolean;
+    };
+  };
+  locale: string;
+  openFirstPaymentMethod: boolean;
+};
+
+type PublisherConfig = {
+  offerId: string;
+  paymentMethods: {
```
Comment (iwonakulacz) [inline src/redux/publisherConfigSlice.ts:37]: I think (not sure if we are doing it already) that we should have there only supported payment methods, so we would have 
```
methodName: 'card' | 'applepay' | 'googlepay' | 'paypal' | 'manual';
paymentGateway:'adyen' | 'paypal' | 'manual'
```

Comment (iwonakulacz) [reply src/redux/publisherConfigSlice.ts:37]: maybe hard to achieve as payment methods as fetched in the old way and the we call `dispatch(updatePaymentMethods(validMethodsFromResponse));` , so also different payment methods could be saved here

Comment (izabelapeza) [reply src/redux/publisherConfigSlice.ts:47]: Okay, I have set methodName and paymentGateway to the above types for now :)

**Pull request closed without merging**